File: test/game.test.js
const { game } = require('../src/js/game');

describe('game.js', () => {
  it('should return 0 if draw', () => {
    let winner = game._getWinner('Rock', 'Rock');
    expect(winner).toBe(0);
  });

  it('should return 1 if Rock is first hand and it wins', () => {
    let winner = game._getWinner('Rock', 'Scissors');
    expect(winner).toBe(1);
  });

  it('should return 2 if Rock is second hand and it wins', () => {
    let winner = game._getWinner('Scissors', 'Rock');
    expect(winner).toBe(2);
  });

  it('should return 1 if Paper is first hand and it wins', () => {
    let winner = game._getWinner('Paper', 'Rock');
    expect(winner).toBe(1);
  });

  it('should return random hand for when a player shoots', () => {
    let randomHand = game._shoot();
    expect(randomHand === 'Rock' || 'Paper' || 'Scissors').toBeTruthy();
  });

});

File: rps-csharp/rps.Tests/GameTest.cs
using System;
using Xunit;

namespace rps.Tests
{
    public class GameTest
    {
        private readonly Game _game;

        public GameTest() 
        {
            _game = new Game();
        }

        [Fact]
        public void shouldReturn0IfDraw()
        {
            var winner = _game.getWinner("Rock", "Rock");
            Assert.True(winner == 0, $"Winner should be 0 instead of {winner}");  
        }

        [Fact]
        public void shouldReturn1IfRockIsFirstAndWins()
        {
            var winner = _game.getWinner("Rock", "Scissors");
            Assert.True(winner == 1, $"Winner should be 1 instead of {winner}");
        }

        [Fact]
        public void shouldReturn2IfRockIsSecondAndWins()
        {
            var winner = _game.getWinner("Scissors", "Rock");
            Assert.True(winner == 2, $"Winner should be 2 instead of {winner}");
        }

        [Fact]
        public void shouldReturn1IfPaperIsFirstAndWins()
        {
            var winner = _game.getWinner("Paper", "Rock");
            Assert.True(winner == 1, $"Winner should be 1 instead of {winner}");
        }

        [Fact]
        public void shouldReturnRandomHandWhenShootIsCalled()
        {
            var randomHand = _game.shoot();
            
            Assert.True(randomHand == "Rock" || 
                        randomHand == "Paper" || 
                        randomHand == "Scissors");
        }
    }
}
